kernel/src/net/socket/netlink/route/kernel/link.rs

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Handle link-related requests.
+
+use core::num::NonZero;
+
+use aster_bigtcp::iface::InterfaceType;
+
+use super::util::finish_response;
+use crate::{
+    net::{
+        iface::{Iface, IFACES},
+        socket::netlink::{
+            message::{CMsgSegHdr, CSegmentType, GetRequestFlags, SegHdrCommonFlags},
+            route::message::{LinkAttr, LinkSegment, LinkSegmentBody, RtnlSegment},
+        },
+    },
+    prelude::*,
+    util::net::CSocketAddrFamily,
+};
+
+pub(super) fn do_get_link(request_segment: &LinkSegment) -> Result<Vec<RtnlSegment>> {
+    let filter_by = FilterBy::from_request(request_segment)?;
+
+    let ifaces = IFACES.get().unwrap();
+    let mut response_segments: Vec<RtnlSegment> = ifaces
+        .iter()
+        // Filter to include only requested links.
+        .filter(|iface| {
+            // `required_index` takes precedence over `required_name`.
```

Comment:
This comment is already in `FilterBy::from_request` and makes no sense here. Better to drop it.